From 321127e23c546269681ae957517e68dc6ea9d24f Mon Sep 17 00:00:00 2001
From: Heinrich Schuchardt <xypron.glpk@gmx.de>
Date: Tue, 17 Oct 2017 18:37:29 +0200
Subject: [PATCH 1/1] efi_loader: use named constant for memory type

When we call an EFI image from memory a memory device path
is created. The memory type should use a named constant.

Fixes: bf19273e81e efi_loader: Add mem-mapped for fallback
Signed-off-by: Heinrich Schuchardt <xypron.glpk@gmx.de>
---
 cmd/bootefi.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/cmd/bootefi.c b/cmd/bootefi.c
index b2f6fd486a..140072f044 100644
--- a/cmd/bootefi.c
+++ b/cmd/bootefi.c
@@ -173,7 +173,7 @@ static unsigned long do_bootefi_exec(void *efi, void *fdt,
 	if (!device_path && !image_path) {
 		printf("WARNING: using memory device/image path, this may confuse some payloads!\n");
 		/* actual addresses filled in after efi_load_pe() */
-		memdp = efi_dp_from_mem(0, 0, 0);
+		memdp = efi_dp_from_mem(EFI_RESERVED_MEMORY_TYPE, 0, 0);
 		device_path = image_path = memdp;
 	} else {
 		assert(device_path && image_path);
-- 
2.14.2

